diff --git a/gcc/jit/jit-recording.cc b/gcc/jit/jit-recording.cc
index 5aae457dc..f99979ad9 100644
--- a/gcc/jit/jit-recording.cc
+++ b/gcc/jit/jit-recording.cc
@@ -2473,7 +2473,9 @@ recording::memento_of_get_type::get_size ()
       size = FLOAT_TYPE_SIZE;
       break;
     case GCC_JIT_TYPE_BFLOAT16:
-      return GET_MODE_UNIT_SIZE (BFmode);
+      // return GET_MODE_UNIT_SIZE (BFmode);
+      size = 16;
+      break;
     case GCC_JIT_TYPE_DOUBLE:
       size = DOUBLE_TYPE_SIZE;
       break;
